src/app/api/attachments/attachments.controller.ts

```diff
@@ -0,0 +1,50 @@
+import { NextRequest, NextResponse } from 'next/server'
+import authenticate from '@api/core/utils/authenticate'
+import { CreateAttachmentRequest, CreateAttachmentRequestSchema } from '@/types/dto/attachments.dto'
+import { AttachmentsService } from '@api/attachments/attachments.service'
+import httpStatus from 'http-status'
+import { handleUpload, type HandleUploadBody } from '@vercel/blob/client'
+import { IdParams } from '@api/core/types/api'
+
+export const createAttachment = async (request: NextRequest) => {
+  const user = await authenticate(request)
+  const body = (await request.json()) as HandleUploadBody
+  const attachmentsService = new AttachmentsService(user)
+  const fileUploadResponse = await handleUpload({
+    body,
+    request,
+    onBeforeGenerateToken: async (pathname: string, clientPayload: string | null) => {
+      if (!user.internalUserId) {
+        throw new Error('Not authorized')
+      }
+      return {
+        tokenPayload: JSON.stringify({
+          data: clientPayload,
+        }),
+      }
+    },
+    onUploadCompleted: async ({ blob, tokenPayload }) => {
+      const data: CreateAttachmentRequest = {
+        taskId: tokenPayload && JSON.parse(JSON.parse(tokenPayload).data).taskId,
```

Comment:
Please use strict types and parsing while using `JSON.parse`. You never know what kind of data you are getting.

Please also use optional chaining if necessary before `.data`